lib/commands/web.js

```diff
@@ -195,25 +195,28 @@ const commands = {
    * @this {XCUITestDriver}
    */
   async getCookies() {
-  if (!this.isWebContext()) {
-    throw new errors.NotImplementedError();
-  }
-
-    this.log.debug('Retrieving all cookies');
+    if (!this.isWebContext()) {
+      throw new errors.NotImplementedError();
+    }
 
-    let script = 'return document.cookie';
-    let jsCookies = await this.executeAtom('execute_script', [script, []]);
+    // get the cookies from the remote debugger, or an empty object
+    const cookies = await this.remote.getCookies() || {cookies: []};
```

Comment:
what `remote.getCookies()` returns if no cookies found?

Reply:
`{"cookies":[]}` in my test, but not sure if it has a case that could return null for example since I haven't tested old versions.

This PR is from the previous deleted code as-is not to change anything right now. (Refactoring would be a separate PR)